src/conversions/sample_rate.rs

```diff
@@ -50,10 +50,14 @@ where
         let from = from.0;
         let to = to.0;
 
+        assert!(num_channels >= 1);
+        assert!(num_channels <= 128);
         assert!(from >= 1);
+        assert!(from <= 500_000);
         assert!(to >= 1);
+        assert!(to <= 500_000);
```

Comment:
Just picked some large numbers, if exceeded the error may only be found by memory allocation errors, which are not helpful. Maybe these can be some global constants so they can be checked separately.

Reply:
I would not insist on this, but in my experience it makes sense to check for obviously unusable values, they may be there by mistake or some malicious misconfiguration.

Reply:
Could you document that in the assert message? Otherwise one might wonder where the number comes from.

Reply:
I decided to remove those for now. Maybe this should be a separate task to make rodio more resilient to misconfiguration, as there are many similar places where earlier diagnostics via assertions can be helpful.

Reply:
> I decided to remove those for now. Maybe this should be a separate task to make rodio more resilient to misconfiguration, as there are many similar places where earlier diagnostics via assertions can be helpful.

That could be useful, as long as they do not impact perf I do not see a disadvantage.